imageregistry/v1/types.go

```diff
@@ -71,7 +73,8 @@ type ImageRegistrySpec struct {
 	// replicas determines the number of registry instances to run.
 	Replicas int32 `json:"replicas" protobuf:"varint,10,opt,name=replicas"`
 	// logging determines the level of logging enabled in the registry.
-	LogLevel int64 `json:"logging" protobuf:"varint,11,opt,name=logging"`
+	// +optional
+	Logging int64 `json:"logging,omitempty" protobuf:"varint,11,opt,name=logging"`
```

Comment:
@ricardomaraschini @dmage is this now a deprecated field, or will it be used to independently set the logging for the registry?

Reply:
For backwards compatibility the idea is to take this field into account if `logLevel` is not set. If both are set, use the "bigger" one.

Reply:
@ricardomaraschini this approach IMO is confusing to admins - we need one field to be the source of truth. I would rather have the following:

1. If `logLevel` > 0, use that and ignore `logging`.
2. If `logLevel` == 0 and `logging` > 0, write the `logging` value to `logLevel` in the spec and proceed.

Because our APIs are forever, we will never remove the `logging` field. Therefore the operator should do the migration on the admin's behalf.

Reply:
We can say that `logging` is deprecated and `logLevel` is a preferred field.

Based on the discussion on Slack:

. | `logLevel` is not set | `logLevel` is set
-|-|-
`logging` is not set | `Normal` (`info`) log level | logLevel is honored
`logging` is set | logging is honored | if logging is `error` or `warning`, logging is honored (as these values cannot be represented by logLevel), otherwise logLevel is honored

Reply:
Added a note informing that `logging` is deprecated and asking to use `logLevel` instead. 

I agree with @adambkaplan and we should automatically migrate values from `logging` into `logLevel` when a conversion 1:1 is possible. The problem stated by @dmage is valid, we don't know if or how many customers are using `error` or `warning`, hopefully we will in the future(https://github.com/openshift/insights-operator/pull/103).

Reply:
@deads2k @adambkaplan @dmage 

There is not much I can do on this side as the migration path will happen on a PR against the Operator. Here we added new a field and tagged the old one as deprecated, nitty-gritty details on migrations should be discussed when a PR comes alive on controller's side.

Could we get this approved/merged?

Reply:
@deads2k @adambkaplan 

The logging field is not an optional today, so it's always set. It can be set to

  * `0` (log only errors)
  * `1` (warning)
  * `2`, `3` (info)
  * anything else (debug)

The normal log level is `2`.

I wouldn't say that `0` is a proper value, because one may forget to set logging in their configuration object and expect the operator to use the default log level.

What are cons of the approach "when logLevel is set, use it, otherwise fallback to logging"?